[Enigmatrix] iP

Enigmatrix opened the pull request:
# Kude
> Subtracting from your list of priorities is as important as adding to it. - [Frank Sonnenberg](https://www.goodreads.com/quotes/tag/to-do-lists)

Kude feels like interacting with an unfeeling system that only cares about getting tasks done.
- Monotonic voice
- Straightforward
- Get ~~shit~~ tasks done quickly.

## How does it work?
1. Use `event` or `deadline` or `todo` to create tasks
2. Use `mark` or `unmark` to set tasks as done/undone
3. `list` or `find` tasks to do

What are you waiting for? Go and `TODO`! 😸 
```java
public class Kude {
    public static void main() {
         // TODO
    }
}
```
- [ ] TODO tick this item

Commit:
Add Gradle support

Enigmatrix committed:
feat: add level 1

Enigmatrix committed:
feat: add, list

Enigmatrix committed:
feat: mark, unmark

Enigmatrix committed:
refactor: add input parser

Enigmatrix committed:
feat: add todo, deadline, event

Enigmatrix committed:
refactor: move models to kude

Enigmatrix committed:
feat: text ui testing

Enigmatrix committed:
feat: delete

Enigmatrix committed:
feat: add serializable models

Enigmatrix committed:
feat: add storage interface

Enigmatrix committed:
feat: save on mutate

Enigmatrix committed:
feat: use datetime instead of string

Enigmatrix committed:
feat: change tests

Enigmatrix committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/kude/tui/Context.java

Enigmatrix committed:
feat: rename Output to Ui and Item* to Task*

Enigmatrix committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Enigmatrix committed:
feat: remove save file for test

Enigmatrix committed:
feat: use proper main class in gradle

Enigmatrix committed:
feat: fix parser line with space end

Enigmatrix committed:
feat: add unit tests

bokung commented on src/main/java/kude/tui/Parser.java:
```diff
@@ -0,0 +1,43 @@
+package kude.tui;
+
+import java.util.*;
```
Here perhaps the wildcard import can be avoided?

bokung commented on src/main/java/kude/tui/Processor.java:
```diff
@@ -0,0 +1,143 @@
+import java.util.HashMap;
+import java.util.Optional;
+
+public class Processor {
```
Some documentation here would be very helpful!

bokung commented on src/main/java/kude/Duke.java:
```diff
@@ -0,0 +1,14 @@
+
+public class Duke {
+
+    public static void main(String[] args) {
```
Very well organised and clean main method!

Enigmatrix replied:
Thanks :)

bokung commented on src/main/java/kude/models/TaskList.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class TaskList implements Serializable {
```
Good use of serialization for saving tasks!

Enigmatrix replied:
thanks :) 😸

bokung commented on src/main/java/kude/tui/Processor.java:
```diff
@@ -0,0 +1,143 @@
+        }
+        this.tasks = tasks;
+    }
+
```
Beautiful organisation of commands!

Enigmatrix replied:
👍

bokung commented on src/main/java/kude/tui/Ui.java:
```diff
@@ -0,0 +1,47 @@
+        out.printf("<%s%s%n", "\t".repeat(indent), line);
+    }
+
+    public void writeErr(String line) {
```
Might be a matter of personal preference here but can consider naming this method `writeError`! It's not much longer and maintains the consistency of using full-length words in this class!

Enigmatrix replied:
👍 🙇

bokung reviewed (commented):
Very well done! Add some java docs and you are good to go!

Enigmatrix committed:
feat: add docs

Enigmatrix committed:
feat: add find command

Enigmatrix committed:
chore: confirm to coding standard

Enigmatrix committed:
Merge branch 'branch-A-CodingStandard'

Enigmatrix committed:
Merge branch 'branch-A-JavaDoc'

Enigmatrix committed:
feat: rename writeErr to writeError

Enigmatrix committed:
feat: add checkstyle config

Enigmatrix committed:
feat: fix checkstyle errs

Enigmatrix committed:
chore: initialize javafx gui

Enigmatrix committed:
feat: initial MainView fxml

Enigmatrix committed:
feat: refactor processor to be more general

Enigmatrix committed:
feat: add dialogbox

Enigmatrix committed:
feat: assert non-null command

Enigmatrix committed:
fix: checkstyle lints

Enigmatrix committed:
feat: remove extranous code

Enigmatrix committed:
feat: replace JavaFX star-imports with individual imports

Enigmatrix committed:
Merge pull request #2 from Enigmatrix/branch-A-Assertions

Add Assertions

Enigmatrix committed:
Merge pull request #3 from Enigmatrix/branch-A-CodeQuality

CodeQuality

Enigmatrix committed:
feat: add place field to models

As per D-Places extension specification, we add the ability
to add a place to any item.

This is done via a new Optional<String> place field in the base class, Task, and initialized via its constructor. The subclasses are updated accordingly.

Enigmatrix committed:
feat: support creation of tasks with places

As per D-Places extension specification, we add the ability
to add a place to any item, and create a task with this place

First of all, Task must be modified to use String instead of Optional<String>
since we want it to be Serializable. We convert it to and from Optional<String>
as necessary.

Next, support for adding a task with a place is added via an optional /place
parameter to event, task and todo commands.

Enigmatrix committed:
Merge pull request #4 from Enigmatrix/branch-D-Places

D-Places Extension

Enigmatrix committed:
feat: add Ui image

Enigmatrix committed:
docs: add ug
